pkg/util/spanlogger/spanlogger.go

```diff
@@ -94,3 +95,23 @@ func (s *SpanLogger) Error(err error) error {
 	s.Span.LogFields(otlog.Error(err))
 	return err
 }
+
+// triggerError calls s.Error if SpanLogger's log level is set to level.ErrorValue().
+func (s *SpanLogger) triggerError(kvps ...interface{}) {
+	var logAsError = false
+	loggedError := errors.New("unknown error")
```

Comment:
No need to allocate it each time. You can initialize it to nil, and then set it to an "unknown error" inside:

```
	if logAsError {
		_ = s.Error(loggedError)
	}
```